src/persist-client/src/internal/gc.rs

```diff
@@ -242,8 +242,9 @@ where
 
         let mut states = machine
             .state_versions
-            .fetch_all_live_states::<K, V, T, D>(&req.shard_id)
+            .fetch_all_live_states::<T>(&req.shard_id)
             .await
+            .check_codecs::<K, V, D>()
```

Comment:
something very interesting here is that we have to specify `<K, V, D>` because nothing actually depends on the types. which is definitely a hint that we could pull the `K, V, D` params out of GC (also makes sense intuitively, given that GC doesn't read or write batch blobs, only deletes them). I'm intentionally not going down this rabbit-hole in this PR though, to try and limit the scariness

Reply:
Yeah, sounds like a nice followup that absolutely doesn't have to happen here. (My suspicion is that a lot of the admin methods ought to be able to switch to less typed versions as well.)

Reply:
Though this looks like the only place that uses your new typed iterator; whenever we move gc over we should make sure to go back and clean that up.

Reply:
Yeah, good call. Added that to the TODO